cli/src/cli.rs

```diff
@@ -703,12 +730,20 @@ fn process_pay(
     timestamp_pubkey: Option<Pubkey>,
     witnesses: &Option<Vec<Pubkey>>,
     cancelable: bool,
+    sign_only: bool,
+    signers: &Option<Vec<(Pubkey, Signature)>>,
+    nonce: Option<Hash>,
 ) -> ProcessResult {
     check_unique_pubkeys(
         (&config.keypair.pubkey(), "cli keypair".to_string()),
         (to, "to".to_string()),
     )?;
-    let (blockhash, fee_calculator) = rpc_client.get_recent_blockhash()?;
+
+    let (blockhash, fee_calculator) = if let Some(nonce) = nonce {
+        (nonce, rpc_client.get_recent_blockhash()?.1)
```

Comment:
If `sign_only` is true, there's a good chance there's no network available right? So, when signing only, you might have to accept a fee from the cli because fee calculator won't be available via RPC

Reply:
Then you must provide a blockhash, no need deal with fees

Reply:
Got it, no fee needed from cli. But this behaviour needs to change when `sign_only == true` so that we don't short circuit here when there's no network

Reply:
If `sign_only == true`, no nonce is provided, and there is no network then that's an error.

Reply:
It looks like this code will error if there is a nonce and `sign_only == true` but there is no network because you are requesting the fee calculator.